### PR TITLE
issue25731 shortcut keys sequence

### PR DESCRIPTION
**Pull request by Malarg**
resolves #25731

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Review by alexeyinkin — CHANGES_REQUESTED** [review]

File: playground/frontend/playground_components/test/src/models/shortcut_test.dart [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+import 'package:playground_components/playground_components.dart';
+
+void main() {
+  group(
+    'BeamShortcut test. ',
+    () {
+      test(
+        'Title builds correctly',
+        () {
+          const meta = 'CMD/CTRL';
```
[the remark this answers is not in the document]
Comment (Malarg) [reply playground/frontend/playground_components/test/src/models/shortcut_test.dart]: fixed on Command. Looks like flutter changes this to Control at linux/windows authomatically

File: playground/frontend/lib/modules/shortcuts/components/shortcuts_modal.dart [hunk 2]
```diff
@@ -94,4 +97,13 @@ class ShortcutsModal extends StatelessWidget {
       ],
     );
   }
+
+  EdgeInsets _buildDialogPadding(BuildContext context) {
```
Comment (alexeyinkin) [inline playground/frontend/lib/modules/shortcuts/components/shortcuts_modal.dart:101]: Fix to ~400px.

File: playground/frontend/lib/modules/shortcuts/components/shortcuts_modal.dart [hunk 3]
```diff
@@ -18,41 +18,28 @@
...
 import 'package:playground/constants/sizes.dart';
 import 'package:playground/modules/shortcuts/components/shortcut_row.dart';
 import 'package:playground/modules/shortcuts/constants/global_shortcuts.dart';
 import 'package:playground_components/playground_components.dart';
 
-const kButtonBorderRadius = 24.0;
-const kButtonWidth = 120.0;
-const kButtonHeight = 40.0;
-const kDialogPadding = 40.0;
+class ShortcutsModalContent extends StatelessWidget {
```
Comment (alexeyinkin) [inline playground/frontend/lib/modules/shortcuts/components/shortcuts_modal.dart:27]:
```suggestion
class ShortcutsDialogContent extends StatelessWidget {
```

File: playground/frontend/lib/modules/shortcuts/components/shortcuts_modal.dart [hunk 4]
```diff
@@ -18,41 +18,28 @@
...
+  static const _kModalMaxWidth = 400.0;
+  static const _kShortcutsMaxWidth = 200.0;
 
-class ShortcutsModal extends StatelessWidget {
-  final PlaygroundController playgroundController;
-
-  const ShortcutsModal({
+  const ShortcutsModalContent({
+    super.key,
```
Comment (alexeyinkin) [inline playground/frontend/lib/modules/shortcuts/components/shortcuts_modal.dart:32]: Delete `key`?

File: playground/frontend/lib/modules/shortcuts/components/shortcuts_modal.dart [hunk 5]
```diff
@@ -18,41 +18,28 @@
...
-      ),
-      contentPadding: const EdgeInsets.all(kDialogPadding),
-      actionsPadding: const EdgeInsets.only(
-        bottom: kDialogPadding,
-        right: kDialogPadding,
-      ),
-      content: Wrap(
+    return ConstrainedBox(
+      constraints: const BoxConstraints(maxWidth: _kModalMaxWidth),
+      child: Wrap(
```
Comment (alexeyinkin) [inline playground/frontend/lib/modules/shortcuts/components/shortcuts_modal.dart:42]: Make vertical, use `spacing`.

File: playground/frontend/lib/pages/standalone_playground/widgets/more_actions.dart [hunk 6]
```diff
@@ -68,9 +68,10 @@ class _MoreActionsState extends State<MoreActions> {
               title: Text(appLocale.shortcuts),
               onTap: () {
                 AnalyticsService.get(context).trackOpenShortcutsModal();
-                showDialog<void>(
+                BeamDialog.show(
```
Comment (alexeyinkin) [inline playground/frontend/lib/pages/standalone_playground/widgets/more_actions.dart:71]: Make a list with buttons.
A CloseButton button likely must take at most a single parameter with a callback to close. See if `PublicNotifier` can be reused here.

**Review by alexeyinkin — CHANGES_REQUESTED** [review]

File: playground/frontend/lib/modules/shortcuts/components/shortcuts_modal.dart (hunk 5 above)
Comment (Malarg) [reply playground/frontend/lib/modules/shortcuts/components/shortcuts_modal.dart:42]: I got issue connected with flutter rendering, also I didn't got problem we a solving by this fix

File: playground/frontend/lib/modules/shortcuts/components/shortcuts_modal.dart (hunk 4 above)
Comment (Malarg) [reply playground/frontend/lib/modules/shortcuts/components/shortcuts_modal.dart]: done

File: playground/frontend/lib/modules/shortcuts/components/shortcuts_modal.dart (hunk 2 above)
Comment (Malarg) [reply playground/frontend/lib/modules/shortcuts/components/shortcuts_modal.dart]: done

File: playground/frontend/playground_components/lib/src/models/shortcut.dart [hunk 7]
```diff
@@ -22,26 +22,28 @@ import 'package:flutter/services.dart';
 import 'intents.dart';
 
 class BeamShortcut {
-  final LogicalKeySet shortcuts;
+  final List<LogicalKeyboardKey> keys;
+  LogicalKeySet get keySet => LogicalKeySet.fromSet(keys.toSet());
```
[the remark this answers is not in the document]
Comment (Malarg) [reply playground/frontend/playground_components/lib/src/models/shortcut.dart:28]: done

File: playground/frontend/lib/modules/shortcuts/components/shortcuts_modal.dart (hunk 5 above)
Comment (alexeyinkin) [reply playground/frontend/lib/modules/shortcuts/components/shortcuts_modal.dart:41]: Use Column, add `alternateWith` or `interleave` function, see https://github.com/dart-lang/collection/issues/227

File: playground/frontend/playground_components/lib/src/widgets/dialog.dart [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+
+  static Future<void> show({
+    required Widget child,
+    required BuildContext context,
+    Widget? title,
+    List<Widget> actions = const [],
+  }) async {
+    await showDialog<void>(
+      context: context,
+      builder: (BuildContext context) => BeamDialog(
```
Comment (alexeyinkin) [inline playground/frontend/playground_components/lib/src/widgets/dialog.dart:41]:
```suggestion
      builder: (BuildContext context) => BeamDialog(
        actions: actions,
        title: title,
        child: child,
      ),
```

**Review by alexeyinkin — CHANGES_REQUESTED** [review]

File: playground/frontend/playground_components/lib/src/widgets/dialog.dart [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import 'package:flutter/material.dart';
+
+const kDialogPadding = 40.0;
```
[the remark this answers is not in the document]
Comment (alexeyinkin) [reply playground/frontend/playground_components/lib/src/widgets/dialog.dart:21]: Make it private if we still need it.

File: playground/frontend/playground_components/assets/translations/en.yaml [hunk 10]
```diff
@@ -43,6 +43,9 @@ widgets:
 
   codeEditor:
     label: 'Code Text Area'
+  
+  closeButton:
+    label: 'CLOSE'
```
[the remark this answers is not in the document]
Comment (Malarg) [reply playground/frontend/playground_components/assets/translations/en.yaml]: done

File: playground/frontend/playground_components/lib/src/widgets/close_button.dart [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import 'package:easy_localization/easy_localization.dart';
+import 'package:flutter/material.dart';
+
+class BeamCloseButton extends StatelessWidget {
+  static const kWidth = 120.0;
+  static const kHeight = 40.0;
```
[the remark this answers is not in the document]
Comment (Malarg) [reply playground/frontend/playground_components/lib/src/widgets/close_button.dart]: done

File: playground/frontend/playground_components/lib/src/widgets/dialog.dart (hunk 9 above)
[the remark this answers is not in the document]
Comment (Malarg) [reply playground/frontend/playground_components/lib/src/widgets/dialog.dart]: I'm sure, we need. Better to make it here than in each dialog. Made private

File: playground/frontend/lib/modules/shortcuts/components/shortcuts_modal.dart
Comment (alexeyinkin) [inline playground/frontend/lib/modules/shortcuts/components/shortcuts_modal.dart:1]: Rename to `shortcuts_dialog.dart`.

**Review by alexeyinkin — APPROVED** [review]
LGTM.

**Comment by Malarg** [comment]
made PR to apache: https://github.com/apache/beam/pull/25876

**Pull request closed without merging**